adagrams/game.py

```diff
@@ -1,11 +1,149 @@
+    'F': 4, 
+    'G': 2, 
+    'H': 4, 
+    'I': 1, 
+    'J': 8, 
+    'K': 5, 
+    'L': 1, 
+    'M': 3, 
+    'N': 1, 
+    'O': 1, 
+    'P': 3, 
+    'Q': 10, 
+    'R': 1, 
+    'S': 1, 
+    'T': 1, 
+    'U': 1, 
+    'V': 4, 
+    'W': 4, 
+    'X': 8, 
+    'Y': 4, 
+    'Z': 10
+}
 
+def draw_letters():
+    letter_freq = {}
+    random_letters=[]
+    selected_letters =[]
+    while len(random_letters) < 26:
+        r = random.choice(list(LETTER_POOL.keys()))
+        random_letters.append(r)
+    for letter in random_letters:
+        if letter in letter_freq:
+            letter_freq[letter] += 1
+        else:
+            letter_freq[letter] = 1
+        if letter_freq[letter] <= LETTER_POOL[letter]:
+            selected_letters.append(letter)
+    return (selected_letters[0:10])
+    
 def uses_available_letters(word, letter_bank):
-    pass
+    word_str = ""
+    temp_word = ""
+    letter_bank_dic = {}
+    word_dic ={}
+    
+    for letter in word:
+        if str.upper(letter) in letter_bank:
+            temp_word += letter
+            word_str += str.upper(letter)
+    for letter in letter_bank:
+        if letter in letter_bank_dic:
+            letter_bank_dic[str.upper(letter)] += 1
+        else:
+            letter_bank_dic[str.upper(letter)] = 1
+    for letter in word_str:
+        if letter in word_dic:
+            word_dic[str.upper(letter)] += 1 
+        else:
+            word_dic[str.upper(letter)] = 1
+    if word_dic[letter] > letter_bank_dic[letter]:
+        return False
+    elif temp_word != word:
+        return False
+    elif temp_word == word:
+        return True   
 
 def score_word(word):
-    pass
+    if word == "":
+        return 0
+    score_word = 0
+    upper_word = str.upper(word)
+    
+    for letter in upper_word:
+        for key, value in SCORE_CHART.items():
+            if letter == key:
+                score_word += value
+                
+    if len(word) in range(7, 11):
+        score_word += 8
+        
+    return score_word 
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    highest_word = []
+    highest_value = 0
+    tie_word = []
+    for word in word_list:
+        word_score = score_word(word)
+        if word_score > highest_value:
+            highest_word = word
+            highest_value = word_score
+            tie_word = [highest_word]
+        elif word_score == highest_value: # tie case
+            tie_word.append(word)
+
+    if len(tie_word) > 1 :
+        if len(tie_word[0]) > len(tie_word[1]) and len(tie_word[0])==10 :
+            return (tie_word[0], highest_value)
+        elif len(tie_word[0]) < len(tie_word[1]) and len(tie_word[1])!=10 :
+            return (tie_word[0], highest_value)
+        elif len(tie_word[0]) == len(tie_word[1]):
+            return(tie_word[0], highest_value)
+        else:
+            return(tie_word[1], highest_value)
+    
+    return (highest_word, highest_value)
+
```

Comment:
Once again, you are creating several extra lists that we don't necessarily need. This works and is a great first pass, but we could definitely pare it down a bit! One implementation would have been to have a tuple that just holds the highest scoring word and the highest score. From there, we can do the following:

1) Make a tuple to hold the first word and its score.
2) For each word in the list, score it and compare it to the score in your highest word tuple.
3) If the score is higher, replace the tuple with the new word.
4) If it is a tie, run the if/else statements you have there toward the bottom of the function!

This really just avoids having to write a tie_word list. We can kind of figure out what the highest word in place.